File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numList = new HashMap<>();
        int[] ans = new int[2];

        for (int i=0;i<nums.length;i++){
            int complement = target - nums[i];
            if (numList.containsKey(complement)){
                ans[0] = numList.get(complement);
                ans[1] = i;
                break;
            }
            numList.put(nums[i], i);
        } 

        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}

        for i in range(len(nums)):
            map[nums[i]] = i

        for i in range(len(nums)):
            lookup_number = target - nums[i]

            if (lookup_number in map) and (map.get(lookup_number) != i):
                return [i, map.get(lookup_number)]